proposals/p1964.md

```diff
@@ -0,0 +1,275 @@
+check and assign the specific value:
+
+```
+fn getSingleEAcuteHex(var combinedAcute: String) -> String {
+    var eAcute: String = "\u{E0}";
+    var accentValue: i32 = 0;
+
+    while (eAcute != combinedAcute) {
+        accentValue = accentValue + 1;
+        eAcute = "\u{E{accentValue}}";
+    }
+    return eAcute;
+}
+
+    var combinedEAcute: String = "\u{65}\u{300}";
+    var a: auto = getSingleEAcuteHex(combinedEAcute);
+```
+
+Using a `String` also leads to further issues regarding familiarty with other
+languages and and leaves the logic generally open to errors, needing further
+logic to check for a valid input.
+
+Further, this design follows other standards set in place by previous proposals.
+For example following the
+[String Literals: Escaping Sequence](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0199.md#escape-sequences-1)
+and representing characters as integers with the behaviour inline with
+[Integer Literals](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0143.md).
+This also supports our goal for
+[Interoperability with and migration from existing C++ code](/docs/project/goals.md#interoperability-with-and-migration-from-existing-c-code)
+by ensuring that every kind of character literal that exists in C++ can be
+represented in a Carbon character literal. This is done in a way that is natural
+to adopt, understand, easy to read by having explicit character types mapped to
+the C++ character types and the correct associated encoding.
+
+## Alternatives considered
+
+### No Distinct Character Literal
+
+In principle a character literal can be represented by reusing string literals.
+//TODO: Add Description
+
+```
+    var b: String = "b";
+    var c: String = "c";
+    if (c > b) {
+        b = c + b;
+    }
+```
+
+We would simply this as a single element string. However it terms of
+readablility, if we had a distinct lexical syntax for character literals versus
+string literals, this would be more inline with Carbon's language design goals
+related to self documenting code, easy to read, understand, write and C++
+interopability.
+
+### Supporting Prefix Declarations
+
+No support is proposed for prefix declarations like `u`, `U`, or `L`. In
+practice they are used to specify the character literal types in languages like
+C and C++, supporting these could help with the familiarity of coming to Carbon
+with a more C++ or C background. Supporting this will help support a much
+cleaner and readable syntax that is more inline with Carbon's goals, i.e
+[Rationale](#rationale).
+
+### Disallowing Numeric Escape Sequences
+
+Not support escaping numeric sequences. This would simplify some of the design
+choices seen above, specifically how we manage [Types](#types) and of the
+`EBCDIC` type, disallowing the `\x...` pattern. This would restrict users by
+having to use the specific hex value, restrict when to use operations between
```

Comment:
@zygoloid  could you explain/what an example of restricting operations on `EBCDIC` types is?

Reply:
What I meant was that it'd make the story for types simpler -- we wouldn't need to worry about whether `'\n'` and `'\x0A'` have the same type if the latter is invalid.

So, I'd say this'd simplify the design by making types simpler and making the restrictions on, and behavior of, operations on character literals simpler.